### ISSUE TITLE
Implement merge commits

### ISSUE THREAD

**Issue by SamuelSchlesinger** [SamuelSchlesinger/version-control#8]
Currently, all commits can technically have more than one previous commit, but when you run `revtool snap` it only adds the one from the branch. We need a command like: `revtool merge <branch>` which will try to bring in the changes from that branch into this one. `revtool rebase <branch>` would be a nice to have as well.

**Comment by SamuelSchlesinger** [comment]
This probably is blocked by #3 because you'll need to resolve diffs in the case of a conflict.

**Comment by SamuelSchlesinger** [comment]
Part of me wants to just do rebases instead? I guess merges are more history-preserving, but I always use rebase in practice.
